### PR TITLE
[All hosts](Script Lab) How to move snippet to yo office

### PR DESCRIPTION
**Pull request by davidchesnut**
New article that shows steps to move your snippet to a standalone project in Yo Office.

### CODE REVIEW

File: docs/overview/create-an-office-add-in-from-script-lab.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+ms.prod: non-product-specific
+description: Learn how to move your snippet from Script Lab into a Yo Office project
+ms.localizationpriority: high
+---
+
+# Create a standalone Office Add-in from your Script Lab code
+
+If you created a snippet in Script Lab, you may want to turn it into a stand-alone add-in. You can copy the code from Script Lab into a a project generated by the Yeoman Generator for Office (also called "Yo Office"). Then you can continue developing the code as an add-in that you can eventually deploy to others.
+
+The steps in this article will refer to [Visual Studio Code](https://code.visualstudio.com/), but you can use any code editor that you prefer.
```
Comment (lindalu-MSFT) [inline docs/overview/create-an-office-add-in-from-script-lab.md:13]:
```suggestion
The steps in this article refer to [Visual Studio Code](https://code.visualstudio.com/), but you can use any code editor that you prefer.
```

File: docs/overview/create-an-office-add-in-from-script-lab.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+- word
+
+> [!IMPORTANT]
+> The `--name` argument value must be in double quotation marks, even if it has no spaces.
+
+The previous command creates a new project folder named **basic-sample**. It is configured to run in the host you specified, and uses TypeScript. Script Lab uses TypeScript by default, but most of the snippets are actually just JavaScript. You can build a Yo Office JavaScript project if you prefer, but just be sure any code you copy over is JavaScript.
+
+## Open the snippet in Script Lab
+
+Let's use an existing snippet in Script Lab to learn how to copy a snippet to a Yo Office generated project.
```
Comment (lindalu-MSFT) [inline docs/overview/create-an-office-add-in-from-script-lab.md:33]:
```suggestion
Use an existing snippet in Script Lab to learn how to copy a snippet to a Yo Office generated project.
```

File: docs/overview/create-an-office-add-in-from-script-lab.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+1. Open Office (Word, Excel, PowerPoint, or Outlook) and then open Script Lab.
+1. Select **Script Lab** > **Code**. If you are working in Outlook, you need to open an email message to read to see Script Lab on the ribbon.
```
Comment (lindalu-MSFT) [inline docs/overview/create-an-office-add-in-from-script-lab.md:36]:
```suggestion
1. Select **Script Lab** > **Code**. If you're working in Outlook, open an email message to see Script Lab on the ribbon.
```

File: docs/overview/create-an-office-add-in-from-script-lab.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+1. Open the **taskpane.html** file, and add the following script tag.
+    
+    ```html
+     <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-3.3.1.js"></script>
+    ```
+    
+    > [!NOTE]
+    > The specific version of jQuery may vary. You can determine which version Script Lab is using by choosing the **Libraries** tab.
+    
+1. Open a terminal in VS Code and enter the following commands
```
Comment (lindalu-MSFT) [inline docs/overview/create-an-office-add-in-from-script-lab.md:84]:
```suggestion
1. Open a terminal in VS Code and enter the following commands.
```

File: docs/overview/create-an-office-add-in-from-script-lab.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    
+    ```command&nbsp;line
+    npm install --save-dev jquery@3.1.1
+    npm install --save-dev @types/jquery@3.3.1
+    ```
+
+If you created a snippet that has additional library dependencies, be sure to add them to the Yo Office project. You can find a list of all library dependencies on the **Libraries** tab in Script Lab.
```
Comment (lindalu-MSFT) [inline docs/overview/create-an-office-add-in-from-script-lab.md:91]:
```suggestion
If you created a snippet that has additional library dependencies, be sure to add them to the Yo Office project. Find a list of all library dependencies on the **Libraries** tab in Script Lab.
```

File: docs/overview/create-an-office-add-in-from-script-lab.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+          $("#run").click(run);
+          });
+    });
+    ```
+
+1. Save the file.
+
+## Custom functions
+
+If your snippet uses custom functions, you need to use the Yo Office custom functions template. Follow these steps to turn you custom functions into a standalone add-in.
```
Comment (lindalu-MSFT) [inline docs/overview/create-an-office-add-in-from-script-lab.md:160]:
```suggestion
If your snippet uses custom functions, you need to use the Yo Office custom functions template. To turn custom functions into a standalone add-in, follow these steps.
```

File: docs/overview/create-an-office-add-in-from-script-lab.md [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+
+1. Run the command `yo office --projectType excel-functions --ts true --name "functions-sample" `.
+    
+    > [!IMPORTANT]
+    > The `--name` argument value must be in double quotation marks, even if it has no spaces.
+    
+1. Open Excel, and then open Script Lab.
+1. Select **Script Lab** > **Code**.
+1. In the Script Lab task pane, choose **Samples**, and then choose the **Basic custom function** sample.
+1. Open the **/src/functions/functions.ts** file. If you are using a JavaScript project, the filename is **functions.js**.
```
Comment (lindalu-MSFT) [inline docs/overview/create-an-office-add-in-from-script-lab.md:170]:
```suggestion
1. Open the **/src/functions/functions.ts** file. If you're using a JavaScript project, the filename is **functions.js**.
```

File: docs/overview/create-an-office-add-in-from-script-lab.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+1. In Script Lab, select the **Script** tab.
+1. Copy all of the code in the **Script** tab to the clipboard. Paste the code at the top of the **functions.ts** (or **functions.js** for javaScript) with the code you copied.
+1. Save the file.
+
+## Test the standalone add-in
+
+Once all the steps are complete you can run and test out your standalone add-in. Run the following command to get started.
```
Comment (lindalu-MSFT) [inline docs/overview/create-an-office-add-in-from-script-lab.md:177]:
```suggestion
Once all the steps are complete, run and test out your standalone add-in. Run the following command to get started.
```

File: docs/overview/create-an-office-add-in-from-script-lab.md [hunk 9]
```diff
@@ -0,0 +1,187 @@
+---
+ms.date: 04/07/2022
+title: Create a standalone Office Add-in from your Script Lab code
+ms.prod: non-product-specific
```
[the remark this answers is not in the document]
Comment (davidchesnut) [reply docs/overview/create-an-office-add-in-from-script-lab.md]: I removed it as most topics don't have it.

[the remark this answers is not in the document]
Comment (lindalu-MSFT) [reply docs/overview/create-an-office-add-in-from-script-lab.md]: It's in the global metadata in the docfx, I believe!

**Review by lindalu-MSFT — APPROVED** [review]

**Comment by lindalu-MSFT** [comment]
@davidchesnut approved!

File: docs/overview/create-an-office-add-in-from-script-lab.md [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+1. In VS Code, open the **/src/taskpane/taskpane.css** file.
+1. In Script Lab, select the **CSS** tab.
+1. Copy all of the CSS in the **CSS**tab to the clipboard. Replace the entire contents of **taskpane.css** with the CSS you copied.
+1. Save all changes to the files you updated in previous steps.
+
+## Add jQuery support
+
+Script Lab uses jQuery in the snippets. You need to add this dependency to the Yo Office project to run the code successfully.
+
+1. Open the **taskpane.html** file, and add the following script tag.
```
[the remark this answers is not in the document]
Comment (alison-mk) [reply docs/overview/create-an-office-add-in-from-script-lab.md:75]: I added it at the end of the file and my add-in works.
```suggestion
1. Open the **taskpane.html** file, and add the following script tag at the end of the file, after the closing `</html>` tag.
```

**Review by alison-mk — APPROVED** [review]
I tested this process and it works! Thanks for putting this together. Including comments and suggestions for a few things I noticed in the article while testing the sample.

**Comment by alison-mk** [comment]
@davidchesnut One more comment: Should we include a **See also** section? Could include links to: 
https://docs.microsoft.com/en-us/office/dev/add-ins/overview/explore-with-script-lab
https://docs.microsoft.com/en-us/office/dev/add-ins/develop/yeoman-generator-overview

File: docs/overview/create-an-office-add-in-from-script-lab.md [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+    $("#save").click(save);
+    ```
+    
+    with:
+    
+    ```typescript
+    Office.onReady(function () {
+      // Office is ready
+      $(document).ready(function () {
+        // The document is ready
```
[the remark this answers is not in the document]
Comment (davidchesnut) [reply docs/overview/create-an-office-add-in-from-script-lab.md:130]: fixed

File: docs/overview/create-an-office-add-in-from-script-lab.md [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+    $("#run").click(run);
+    ```
+    
+    with:
+    
+    ```typescript
+    Office.onReady(function () {
+      // Office is ready
+      $(document).ready(function () {
+        // The document is ready
```
[the remark this answers is not in the document]
Comment (davidchesnut) [reply docs/overview/create-an-office-add-in-from-script-lab.md:150]: fixed

File: docs/overview/create-an-office-add-in-from-script-lab.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (davidchesnut) [reply docs/overview/create-an-office-add-in-from-script-lab.md]: It's best in the `<head>` tag along with the other script tags, although I suppose in this case it does work at the bottom of the file.

[the remark this answers is not in the document]
Comment (davidchesnut) [reply docs/overview/create-an-office-add-in-from-script-lab.md]: i updated to indicate to use the head tag.

**Pull request merged**